services/bettersocial/post/createPostV3.js

```diff
@@ -86,12 +86,16 @@ function getPollPostExpiredAt(pollsDuration, durationFeed) {
 }
 
 async function processPost({userId, isAnonimous, body, data, filteredTopics}) {
+  const expiredAt =
+    body.duration_feed !== 'never'
+      ? moment().utc().add(+body.duration_feed, 'day')
+      : moment().utc().add(100, 'years');
   const [newPost] = await Promise.all([
     Post.create({
       post_id: uuid(),
       author_user_id: userId,
       anonymous: isAnonimous,
-      duration: moment().utc().add(+body.duration_feed, 'day'),
+      duration: expiredAt,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here seem fine. The `duration` field is now being set to the calculated `expiredAt` value which makes sense given the context provided. However, please ensure that the `Post.create` function and the database schema can handle this new type of input for the `duration` field. If the `duration` field was previously expected to be a number representing days, and now it's a date, this could potentially cause issues.


<!-- This is an auto-generated comment by OSS CodeRabbit -->